internal/dataplane/kong_client.go

```diff
@@ -387,6 +386,20 @@ func (c *KongClient) Update(ctx context.Context) error {
 	c.lock.Lock()
 	defer c.lock.Unlock()
 
+	// If Kong is running in dbless mode, we can fetch and store the last good configuration.
+	if c.dbmode == "" || c.dbmode == "off" {
+		// Fetch the last valid configuration from the proxy only in case there is no valid
+		// configuration already stored in memory. This can happen when KIC restarts and there
+		// already is a Kong Proxy with a valid configuration loaded.
+		if c.lastValidKongState == nil {
+			if err := c.FetchLastGoodConfiguration(ctx); err != nil {
+				// If the client fails to fetch the last good configuration, we log it
+				// and carry on, as this is a condition that can be recovered with the following steps.
+				c.logger.Error(err)
```

Comment:
I'd suggest giving the log entry more context:
```suggestion
				c.logger.WithError(err).Error("failed to fetch last good configuration from the proxy")
```